integrations/adobe-analytics/lib/index.js

```diff
@@ -292,19 +293,21 @@ AdobeAnalytics.prototype.track = function(track) {
   // Delete any existing keys on window.s from previous call
   clearKeys(dynamicKeys);
 
+  var eventName = track.event().toLowerCase();
+
   // Map to Heartbeat events if enabled.
   if (this.options.heartbeatTrackingServerUrl) {
-    var heartbeatFunc = this.heartbeatEventMap[track.event().toLowerCase()];
+    var heartbeatFunc = this.heartbeatEventMap[eventName];
     if (heartbeatFunc) {
       heartbeatFunc.call(this, track);
       return; // Heartbeat calls AA itself, so returning here likely prevents dupe data.
     }
   }
-
-  // Find AA event name from setting's event map
-  // otherwise abort
-  var adobeEvent = aliasEvent(track.event(), this.options.events);
-  if (!adobeEvent) return;
+  // Check if Segment event is mapped in settings; if not, noop
+  var isMapped = this.isMapped(eventName);
+  if (!isMapped) {
+    return;
+  }
 
   this.processEvent(track);
```

Comment:
```suggestion
```suggestion
  // 'cart opened' is not covered in our ecommerce spec and was one-off specced for this integration
if (eventName === 'cart opened' || eventName === 'opened cart') {
  this.processEvent(track, 'scOpen');
} else {
  this.processEvent(track);
};
```

Before we release all these changes GA can we commit this change? This will add support for Cart Opened which was raised by Levis. [I see in our SS code](https://github.com/segmentio/integrations/blob/a626606b319defa9169795651fb7f808887bddd3/integrations/adobe-analytics/lib/mapper.js#L45-L50) we added this as one off support since Cart Opened isn't part of our ecommerce spec. This relates to this ticket that was raised by Levis: https://segment.atlassian.net/browse/STRATCONN-178

Reply:
That makes sense to me. Could we commit it tomorrow morning (05/05) and pin Levi's source to this new version? At least we could (a) solve the issue (b) test this mashup of all the changes.